common/src/main/java/com/revenuecat/purchases/common/verification/verificationResultExtensions.kt

```diff
@@ -0,0 +1,15 @@
+package com.revenuecat.purchases.common.verification
+
+import com.revenuecat.purchases.VerificationResult
+
+fun VerificationResult.compareAndReturnMostRestrictiveResult(
+    verificationResult: VerificationResult
+): VerificationResult {
+    return if (this == VerificationResult.FAILED) {
+        // These situations can only happen if using `informational` verification mode.
+        // When using `enforced` we won't cache verification errors.
+        VerificationResult.FAILED
+    } else {
+        verificationResult
```

Comment:
There are other situations involving `NOT_REQUESTED` but those shouldn't happen once we clear the cache after enabling entitlement verification